rust-crates/symblib/src/gosym/mod.rs

```diff
@@ -540,3 +540,55 @@ fn range_rel2abs(base: VirtAddr, rng: Range<raw::TextStartOffset>) -> Range<Virt
         end: base.wrapping_add(rng.end.0),
     }
 }
+
+impl symbconv::PointResolver for GoRuntimeInfo<'_> {
+    fn symbols_for_pc(&self, pc: VirtAddr) -> symbconv::Result<Vec<symbconv::ResolvedSymbol>> {
+        let mut symbols = Vec::new();
+
+        match self.find_func(pc) {
+            Ok(Some(func)) => {
```

Comment:
Nit: I'd flatten this to something like:

```
let func = match self.find_func(pc) {
    Ok(Some(func)) => func,
    Ok(None) return Ok(Vec::new()),
    Err(e) => return Err(symbconv::Error::Go(symbconv::go::Error::Gosym(e))),
};

// code that was previously in the `Ok(Some(x))` arm here
```

Reply:
resolved with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/380/commits/a73f6f80777d80375c5e90d314ef484751185569